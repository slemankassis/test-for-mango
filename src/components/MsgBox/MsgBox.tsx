import React, { FC } from 'react';
import styled from 'styled-components';

const ColorBox = styled.div`
  color: #fff;
  padding: 0.5em 0.5em 0.3em;

  &.m-error {
    background-color: #ff4444;
    box-shadow: 0 8px #cc0000;
  }

  &.m-info {
    background-color: #33b5e5;
    box-shadow: 0 8px #0099cc;
  }
`;

enum MsgBoxType {
  error = 'm-error',
  info = 'm-info',
}

export interface MsgBoxProps {
  type: keyof typeof MsgBoxType;
}

const MsgBox: FC<MsgBoxProps> = ({ children, type }) => {
  if (children) {
    return <ColorBox className={MsgBoxType[type]}>{children}</ColorBox>;
  } else {
    return null;
  }
};

export default MsgBox;
