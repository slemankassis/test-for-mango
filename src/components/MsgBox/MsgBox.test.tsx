import React from 'react';
import { render } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import MsgBox, { MsgBoxProps } from './MsgBox';

const setup = (msgBoxProps: MsgBoxProps) => {
  const utils = render(<MsgBox {...msgBoxProps}>test</MsgBox>);

  return {
    el: utils.getByText('test'),
    ...utils,
  };
};

describe('MsgBox check props', () => {
  test('should have class info', () => {
    const { el } = setup({ type: 'info' });

    expect(el).toHaveClass('m-info');
  });

  test('should have class error', () => {
    const { el } = setup({ type: 'error' });

    expect(el).toHaveClass('m-error');
  });

  test('should not exist when empty', () => {
    const { container } = render(<MsgBox type="info"></MsgBox>);

    expect(container.firstChild).toBe(null);
  });
});

describe('snapshots MsgBox', () => {
  test('snapshot empty error match', () => {
    const tree = TestRenderer.create(<MsgBox type="error" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('snapshot empty info match', () => {
    const tree = TestRenderer.create(<MsgBox type="info" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('snapshot full error match', () => {
    const tree = TestRenderer.create(<MsgBox type="error">Test</MsgBox>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('snapshot full error match', () => {
    const tree = TestRenderer.create(<MsgBox type="info">Test</MsgBox>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
