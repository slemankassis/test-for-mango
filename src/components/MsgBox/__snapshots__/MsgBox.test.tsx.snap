// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots MsgBox snapshot empty error match 1`] = `null`;

exports[`snapshots MsgBox snapshot empty info match 1`] = `null`;

exports[`snapshots MsgBox snapshot full error match 1`] = `
<div
  className="sc-bdfBwQ cjQMhH m-error"
>
  Test
</div>
`;

exports[`snapshots MsgBox snapshot full error match 2`] = `
<div
  className="sc-bdfBwQ cjQMhH m-info"
>
  Test
</div>
`;
